src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfoOfT.cs

```diff
@@ -21,5 +21,18 @@ internal JsonTypeInfo()
         {
             Debug.Assert(false, "This constructor should not be called.");
         }
+
+        /// <summary>
+        /// A method that serializes an instance of <typeparamref name="T"/> using
+        /// <see cref="JsonSerializerOptionsAttribute"/> values specified at design time.
+        /// </summary>
+        public Action<Utf8JsonWriter, T>? Serialize { get; internal set; }
+
+        /// <summary>
+        /// When serializing objects with source-gen converters, indicates whether the
+        /// configured <see cref="JsonSerializerOptions"/> is compatible with the generated
+        /// <see cref="JsonTypeInfo{T}.Serialize"/> func.
+        /// </summary>
+        internal bool UseFastPathOnWrite { get; set; }
```

Comment:
I don't think I'd call this `FastPath` in the code. Instead I would tie it to the `Serialize` action. `UseSerializeDelegateOnWrite` or similar.